cache/disk/disk.go

```diff
@@ -169,7 +181,7 @@ func New(dir string, maxSizeBytes int64, opts ...Option) (*Cache, error) {
 		f := filepath.Join(dir, c.FileLocation(kind, value.legacy, hash, value.size, value.random))
 
 		// Run in a goroutine so we can release the lock sooner.
-		go removeFile(f)
+		go c.removeFile(f)
```

Comment:
Won't this still start unboundedly many go-routines and then ensures that we don't delete more than 5000 files concurrently?

Reply:
Yes, but I don't think having a large number of goroutines is a problem here. It's only the OS thread limit that we need to avoid hitting.